src/FrenetOptimalTrajectory/FrenetOptimalTrajectory.cpp

```diff
@@ -31,44 +35,230 @@ FrenetOptimalTrajectory::FrenetOptimalTrajectory(
     csp = new CubicSpline2D(x, y);
 
     // calculate the trajectories
-    calc_frenet_paths();
+    if (fot_hp->num_threads == 0) {
+        calc_frenet_paths();
+    } else { // if threading
+        vector<thread> threads;
+
+        // calculate how to split computation across threads
+        int num_di_iter = static_cast<int>(
+            (fot_hp->max_road_width_l + fot_hp->max_road_width_r) /
+            fot_hp->d_road_w);
+        num_di_iter = num_di_iter + 1; // account for the last index
+
+        int iter_di_index_range =
+            static_cast<int>(num_di_iter / fot_hp->num_threads);
+
+        for (int i = 0; i < fot_hp->num_threads; i++) {
+            if (i != fot_hp->num_threads - 1) {
+                threads.push_back(thread(
+                    &FrenetOptimalTrajectory::calc_frenet_paths_threaded, this,
+                    i * iter_di_index_range, (i + 1) * iter_di_index_range));
+            } else { // account for last thread edge case
+                threads.push_back(
+                    thread(&FrenetOptimalTrajectory::calc_frenet_paths_threaded,
+                           this, i * iter_di_index_range, num_di_iter));
+            }
+        }
+
+        // wait for all threads to finish computation
+        for (auto &t : threads) {
+            t.join();
+        }
+    }
```

Comment:
Can you please move this code into a threaded_calc_all_frenet_paths method?

Reply:
Sure, that's cleaner. Thanks!